crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -199,28 +199,37 @@ impl Timeline {
                 }))
                 .await?;
 
-            pagination_outcome.events_received = messages
-                .chunk
-                .len()
-                .try_into()
-                .expect("Pagination response can't have so >65535 events");
-            pagination_outcome.total_events_received += pagination_outcome.events_received;
-            pagination_outcome.items_added = 0;
-            pagination_outcome.items_updated = 0;
-
-            for room_ev in messages.chunk {
-                let res = self.inner.handle_back_paginated_event(room_ev, own_user_id).await;
-                pagination_outcome.items_added += res.item_added as u16;
-                pagination_outcome.total_items_added += res.item_added as u16;
-            }
+            from = messages.end;
 
-            pagination_outcome.total_items_added += pagination_outcome.items_added;
-            pagination_outcome.total_items_updated += pagination_outcome.items_updated;
+            let process_events = async {
+                outcome.events_received = messages.chunk.len().try_into().ok()?;
+                outcome.total_events_received =
+                    outcome.total_events_received.checked_add(outcome.events_received)?;
+                outcome.items_added = 0;
+                outcome.items_updated = 0;
+
+                for room_ev in messages.chunk {
+                    let res = self.inner.handle_back_paginated_event(room_ev, own_user_id).await;
+                    outcome.items_added = outcome.items_added.checked_add(res.item_added as u16)?;
+                    outcome.items_updated = outcome.items_updated.checked_add(res.items_updated)?;
+                }
+
+                outcome.total_items_added =
+                    outcome.total_items_added.checked_add(outcome.items_added)?;
+                outcome.total_items_updated =
+                    outcome.total_items_updated.checked_add(outcome.items_updated)?;
+
+                Some(())
+            };
 
-            from = messages.end;
             if from.is_none() {
                 break;
             }
+
+            if process_events.await.is_none() {
+                error!("u16 overflow");
```

Comment:
Let's make the message a bit clearer:

1. Why and where is there an u16 overflow
2. What are the implications of such an overflow

Something like "the server sent us more events than we can handle" or "the server sent us too many events, giving up".